src/stan/model/indexing/rvalue.hpp

```diff
@@ -262,21 +269,23 @@ rvalue(
  * @param[in] depth Depth of indexing dimension.
  * @return Result of indexing matrix.
  */
-template <typename T, typename I>
-inline typename boost::disable_if<boost::is_same<I, index_uni>,
-                                  Eigen::Matrix<T, Eigen::Dynamic, 1> >::type
-rvalue(
-    const Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>& a,
-    const cons_index_list<I, cons_index_list<index_uni, nil_index_list> >& idx,
+template <typename EigMat, typename I,
+          typename = require_not_same_t<index_uni, I>,
+          typename = require_eigen_t<EigMat>,
+          typename = require_not_eigen_vector_t<EigMat>>
+inline auto rvalue(
+    const EigMat& a,
+    const cons_index_list<I, cons_index_list<index_uni, nil_index_list>>& idx,
     const char* name = "ANON", int depth = 0) {
   int rows = rvalue_index_size(idx.head_, a.rows());
-  Eigen::Matrix<T, Eigen::Dynamic, 1> c(rows);
+  Eigen::Matrix<scalar_type_t<EigMat>, EigMat::RowsAtCompileTime, 1> c(rows);
```

Comment:
We know this is dynamic due to `require_not_eigen_vector_t<EigMat>`.
```suggestion
  Eigen::Matrix<scalar_type_t<EigMat>, Eigen::Dynamic, 1> c(rows);
```

Reply:
I can change this, the only reason I can think of to use RowsAtCompileTime here is if we started allowed for fixed size matrices